From 8a4fb4b9804c9a3fe9862db0e38a920e9c8c77b4 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sun, 10 Nov 2019 08:22:55 +0200
Subject: [PATCH] [SW-6739] arcboot: hex create script after build

add script after build has finished to create hex file image

Change-Id: I68e0cd8d34946ce668271beb91ebcb6eb3383b2b
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-arcboot.sh b/board/habanalabs/common/post-image-arcboot.sh
new file mode 100755
index 0000000..470fcad
--- /dev/null
+++ b/board/habanalabs/common/post-image-arcboot.sh
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+pushd ${BINARIES_DIR}
+	# Convert bin file into Palladium hex format
+	xxd -ps -c4					arcboot.bin	 		 > arcboot.hex
+popd
diff --git a/configs/goya2_arc_defconfig b/configs/goya2_arc_defconfig
index 99f650a..e74e17f 100644
--- a/configs/goya2_arc_defconfig
+++ b/configs/goya2_arc_defconfig
@@ -1,5 +1,6 @@
 BR2_arcle=y
 BR2_archs46=y
+BR2_ROOTFS_POST_BUILD_SCRIPT="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/post-image-arcboot.sh"
 BR2_SVN="svn"
 BR2_DL_DIR="$(BR2_EXTERNAL_HABANALABS_PATH)/dl"
 BR2_PACKAGE_OVERRIDE_FILE="$(BR2_EXTERNAL_HABANALABS_PATH)/board/habanalabs/common/local.mk"
